From 189b430d382eacfd2b9cf64be1464dfcb913b8a7 Mon Sep 17 00:00:00 2001
From: Chun-wei Fan <fanchunwei@src.gnome.org>
Date: Mon, 18 Aug 2014 12:27:28 +0800
Subject: [PATCH] rsvg-base.c: Have realpath() Work On Windows

Use _fullpath() on Windows to do the job for realpath() as realpath() does
not exist on Windows.

https://bugzilla.gnome.org/show_bug.cgi?id=710163
---
 rsvg-base.c | 9 +++++++++
 1 file changed, 9 insertions(+)

diff --git a/rsvg-base.c b/rsvg-base.c
index 683d607..e3818d7 100644
--- a/rsvg-base.c
+++ b/rsvg-base.c
@@ -58,6 +58,15 @@
 #include "rsvg-xml.h"
 
 /*
+ * XXX: Perhaps do a GIO-based implementation for
+ * realpath() or use gnulib implementation for this
+ * https://bugzilla.gnome.org/show_bug.cgi?id=710163
+ */
+#ifdef G_OS_WIN32
+#define realpath(a,b) _fullpath(b,a,_MAX_PATH)
+#endif
+
+/*
  * This is configurable at runtime
  */
 #define RSVG_DEFAULT_DPI_X 90.0
-- 
1.8.3.msysgit.0

